app/services/arcgis_api/geocoder.rb

```diff
@@ -1,5 +1,9 @@
 module ArcgisApi
   class Geocoder
+    def initialize(&block)
+      @custom_connection_options = block || proc {}
```

Comment:
Let's leave the variable nil if it's not there
```suggestion
      @custom_connection_options = block
```